src/frontend/src/features/layouts/components/admin/modal-create-address/index.tsx

```diff
@@ -131,6 +134,16 @@ export const ModalCreateAddress = ({ isOpen, onClose }: ModalCreateAddressProps)
     }
   }, [watchedValues, activeTab, prefixManuallyChanged, setValue]);
 
+  // Focus first field when tab changes
+  useEffect(() => {
+    if (firstFieldRef && isOpen) {
+      // Small delay to ensure the field is rendered
+      setTimeout(() => {
+        firstFieldRef.focus();
+      }, 100);
+    }
+  }, [activeTab, isOpen, firstFieldRef]);
```

Comment:
```suggestion
  useEffect(() => {
    if (activeTab === 'personal') form.setFocus('first_name');
    else if (activeTab === 'shared') form.setFocus('name');
    else if (activeTab === 'redirect') form.setFocus('prefix');
  }, [form.setFocus, activeTab]);
```

In this way you should be able to remove the field ref and the hacky timeout.